src/createReducer.ts

```diff
@@ -97,15 +116,21 @@ export function createReducer<S>(
   initialState: S,
   mapOrBuilderCallback:
     | CaseReducers<S, any>
-    | ((builder: ActionReducerMapBuilder<S>) => void)
+    | ((builder: ActionReducerMapBuilder<S>) => void),
+  actionMatchers: ActionMatcherTupleCollection<S> = []
 ): Reducer<S> {
-  let actionsMap =
+  let [actionsMap, builderActionMatchers = []] =
     typeof mapOrBuilderCallback === 'function'
       ? executeReducerBuilderCallback(mapOrBuilderCallback)
-      : mapOrBuilderCallback
+      : [mapOrBuilderCallback]
 
   return function(state = initialState, action): S {
-    const caseReducer = actionsMap[action.type]
+    const caseReducer =
+      actionsMap[action.type] ||
+      actionMatchers
+        .concat(builderActionMatchers)
```

Comment:
Silly micro-optimization: can we do the  `concat` once outside of the returned function?

Reply:
Good micro-point. Will do.